src/main/resources/api/data-repository-openapi.yaml

```
@@ -2828,6 +2828,51 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/datasets/{id}/enumerateSnapshotRequests:
```

I think the REST convention is to call this endpoint `snapshotRequests`; that's the convention TDR users for jobs, snapshots, datasets, etc.
```suggestion
  /api/repository/v1/datasets/{id}/snapshotRequests:
```

---

Similarly, I wonder if `createSnapshotRequest` should really be a `post` on the same endpoint (`/api/repository/v1/datasets/{id}/snapshotRequests`)? That would also be the REST-ful way to do it.

---

Okay that makes sense. I will refactor this and make a UI PR to match the new endpoint